server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/BaseStarTreeBuilder.java

```diff
@@ -112,19 +114,25 @@ protected BaseStarTreeBuilder(
 
         this.starTreeField = starTreeField;
         StarTreeFieldConfiguration starTreeFieldSpec = starTreeField.getStarTreeConfig();
-
-        List<Dimension> dimensionsSplitOrder = starTreeField.getDimensionsOrder();
-        this.numDimensions = dimensionsSplitOrder.size();
+        int numDims = 0;
+        for (Dimension dim : starTreeField.getDimensionsOrder()) {
+            numDims += dim.getNumSubDimensions();
+            dimensionsSplitOrder.add(dim);
```

Comment:
should we add a comment on where dimensionSplitOrder is defined, that we would care about extrapolated dimensions from parent dimensions.

Reply:
There is no extropolation, some dimensions like date does have multiple inputs which is represented in the same order here.